### ISSUE TITLE
Initial implementation of metrics -> BigQuery loader

### ISSUE THREAD

**Issue by Riaankl** [kubernetes/k8s.io#2045]
We would like to further progress on the good work done by @justinsb here:
https://github.com/justinsb/k8s.io/commit/9f64ea7540bc28ae03bc0b2b42c4da27fb777da4

/cc @hh

**Comment by BobyMCbobs** [comment]
`./create-schema.sh` seems to have run successfully. The schema and table exist in BigQuery.
```
$ ./create-schema.sh                                                                                                                                                                                         
+ gcloud config get-value project
+ PROJECT=k8s-infra-ii-sandbox
+ bq mk --location=US --dataset k8s-infra-ii-sandbox:kubernetes_public_logs
Dataset 'k8s-infra-ii-sandbox:kubernetes_public_logs' successfully created.
+ bq mk --table --description Raw logs from GCS --label pii:yes --norequire_partition_filter --time_partitioning_field request_time --time_partitioning_type DAY k8s-infra-ii-sandbox:kubernetes_public_logs.raw_gcs_logs schemas/raw_gcs_logs
Table 'k8s-infra-ii-sandbox:kubernetes_public_logs.raw_gcs_logs' successfully created.
```

There may be something wrong with how I'm running [loader](https://github.com/justinsb/k8s.io/blob/9f64ea7540bc28ae03bc0b2b42c4da27fb777da4/metrics/access-logs/loader/main.go), the final line seems to be to do with GCS auth.
```
$ cd loader
$ make
$ go build -o ./bin/loader .
$ ./bin/loader --project $(gcloud config get-value project)
[REDACTED OUTPUT]
storage.NewClient failed: dialing: google: could not find default credentials. See https://developers.google.com/accounts/docs/application-default-credentials for more information.
```

The output shows all of the cloud-provider CIDRs as expected.
However, no data appears populated in the BigQuery table.

**Comment by justinsb** [comment]
Hmmm, well the answer should be to run `gcloud auth application-default login` and login as yourself (with the browser flow).  I assume because `bq` worked that you've already run `gcloud auth login`.

Questions for me (but not for this PR!):

* Why did that link not show anything useful?
* Is there some way to not need application-default credentials?

Anyway ... can you let me know if `gcloud auth application-default login` works / gets you past authentication...

**Comment by justinsb** [comment]
Oh, and the way you're running it looks good.  You can also just `cd loader; go run . --project $(gcloud config get-value project)` (I think!)

**Comment by BobyMCbobs** [comment]
> Hmmm, well the answer should be to run `gcloud auth application-default login` and login as yourself (with the browser flow). I assume because `bq` worked that you've already run `gcloud auth login`.
> 
> Questions for me (but not for this PR!):
> 
> * Why did that link not show anything useful?
> * Is there some way to not need application-default credentials?
> 
> Anyway ... can you let me know if `gcloud auth application-default login` works / gets you past authentication...

This appears to unblock it!

Now I'm getting the logs:
```
I0514 15:10:21.475011  346158 main.go:111] scanning for objects
I0514 15:10:22.749774  346158 main.go:147] loading gs://k8s-artifacts-gcslogs/[FILE PATH HERE]e7a4d9ac_v0
...
```

I'm sure it will take a bit of time to run, but this is good!

**Comment by BobyMCbobs** [comment]
OK, output shows this message now.

```
W0514 15:10:41.517401  346158 main.go:125] continuing after error: loadObject("gs://k8s-artifacts-gcslogs/[FILE PATH HERE]e7a4d9ac_v0") failed: Send to row stream failed: EOF
```

Perhaps something with the upload being cut off?

**Comment by justinsb** [comment]
Yes, probably just a network blip.  It shouldn't matter in the grand scheme of things.  The way it works is that we process each GCS file one at a time.  After we upload a file into BigQuery, we mark some metadata in GCS to indicate that we've uploaded it.  5 minutes later we run again and look for new files / files that had an error and weren't marked as done.  So probably it'll clear itself up in the next cycle.  Given it was an EOF, I'm assuming a transient network blip, but if it keeps happening then we can dig in..

There is a small window where after we've uploaded into BigQuery, if we crash or otherwise fail to mark the metadata, then we might end up double-importing the file.  I'd argue it doesn't really matter in the big picture, as long as it doesn't happen "too often"

**Comment by BobyMCbobs** [comment]
@justinsb, I've just given _loader_ a go again.
I seem to be experiencing the same errors.

```
I0517 13:31:48.640229  323604 main.go:111] scanning for objects
I0517 13:31:49.753237  323604 main.go:147] loading gs://k8s-artifacts-gcslogs/asia.artifacts.k8s-artifacts-prod.appspot.com_usage_2021_05_02_07_00_00_003d75f427e7a4d9ac_v0
W0517 13:32:08.332199  323604 main.go:125] continuing after error: loadObject("gs://k8s-artifacts-gcslogs/asia.artifacts.k8s-artifacts-prod.appspot.com_usage_2021_05_02_07_00_00_003d75f427e7a4d9ac_v0") failed: Send to row stream failed: EOF
I0517 13:32:08.332214  323604 main.go:147] loading gs://k8s-artifacts-gcslogs/asia.artifacts.k8s-artifacts-prod.appspot.com_usage_2021_05_02_07_00_00_013d75f427e7a4d9ac_v0
W0517 13:32:25.083002  323604 main.go:125] continuing after error: loadObject("gs://k8s-artifacts-gcslogs/asia.artifacts.k8s-artifacts-prod.appspot.com_usage_2021_05_02_07_00_00_013d75f427e7a4d9ac_v0") failed: Send to row stream failed: EOF
I0517 13:32:25.083031  323604 main.go:147] loading gs://k8s-artifacts-gcslogs/asia.artifacts.k8s-artifacts-prod.appspot.com_usage_2021_05_02_07_00_00_023d75f427e7a4d9ac_v0
W0517 13:32:42.055040  323604 main.go:125] continuing after error: loadObject("gs://k8s-artifacts-gcslogs/asia.artifacts.k8s-artifacts-prod.appspot.com_usage_2021_05_02_07_00_00_023d75f427e7a4d9ac_v0") failed: Send to row stream failed: EOF
I0517 13:32:42.055056  323604 main.go:147] loading gs://k8s-artifacts-gcslogs/asia.artifacts.k8s-artifacts-prod.appspot.com_usage_2021_05_02_07_00_00_033d75f427e7a4d9ac_v0
W0517 13:33:03.245307  323604 main.go:125] continuing after error: loadObject("gs://k8s-artifacts-gcslogs/asia.artifacts.k8s-artifacts-prod.appspot.com_usage_2021_05_02_07_00_00_033d75f427e7a4d9ac_v0") failed: Send to row stream failed: EOF
I0517 13:33:03.245323  323604 main.go:147] loading gs://k8s-artifacts-gcslogs/asia.artifacts.k8s-artifacts-prod.appspot.com_usage_2021_05_02_07_00_00_043d75f427e7a4d9ac_v0
...
```

Do you have any thoughts on this?

**Comment by spiffxp** [comment]
/wg k8s-infra
is this still relevant?

**Comment by ameukam** [comment]
@BobyMCbobs What's the current state about this ?

**Comment by BobyMCbobs** [comment]
> /wg k8s-infra is this still relevant?

> @BobyMCbobs What's the current state about this ?

@spiffxp @ameukam, I believe this is superseded by [public-log-asn-matcher](https://github.com/kubernetes/k8s.io/tree/853d1e1/images/public-log-asn-matcher)
